easybuild/easyconfigs/j/Java/Java-15.0.2.eb

```diff
@@ -0,0 +1,19 @@
+#
+# This file is an EasyBuild reciPY as per https://github.com/easybuilders/easybuild
+#
+# Author:    Robert Mijakovic <robert.mijakovic@lxp.lu>
+#
+name = 'Java'
+version = '15.0.2'
+
+homepage = 'https://openjdk.java.net'
+description = """Java Platform, Standard Edition (Java SE) lets you develop and deploy
+ Java applications on desktops and servers."""
+
+toolchain = SYSTEM
+
+source_urls = ['https://download.java.net/java/GA/jdk15/779bf45e88a44cbd9ea6621d33e33db1/36/GPL/']
```

Comment:
The `source_urls` and `sources` give me a download location of `https://download.java.net/java/GA/jdk15/779bf45e88a44cbd9ea6621d33e33db1/36/GPL/openjdk-15.0.2_linux-x64_bin.tar.gz` but that fails.

Reply:
Yes, sorry, I need to remove the download link. Oracle generates the hash in between and it should be downloaded manually.

Reply:
The only semi-solution I found is to grep the web page where they allow download:
`curl -s https://www.oracle.com/java/technologies/javase-jdk15-downloads.html | grep "otn-pub.*linux.*tar.gz.*x64" | sed -e "s/.*data-file='/https:/g" -e "s/' data-license.*//g"`
or
`curl -s https://jdk.java.net/15/ | grep "linux-x64.*\.sha256" | sed -e "s/.*href=\"//g" -e "s/\.sha256.*//g"`

None of them actually points to 15.0.2 but rather to the newest bugfix of release 15.

Reply:
Also, looking into it once again, by adding extending the java easyblock it could handle multiple Java releases. For the moment, aarch64 and x64.

```
from easybuild.tools.build_log import EasyBuildError
from easybuild.tools.systemtools import X86_64, AARCH64, get_cpu_architecture

class EB_Java(PackedBinary):                                                          
...
    def __init__(self, *args, **kwargs):
        """ Init the Java easyblock adding a new javaarch template var """
        myarch = get_cpu_architecture()
        if myarch == X86_64:
            javaarch = 'x64'
        elif myarch == AARCH64:
            javaarch = 'aarch64'
        else:                                                                         
            raise EasyBuildError("Architecture %s is not supported for Java on EasyBuild", myarch)

        super(EB_cuDNN, self).__init__(*args, **kwargs)                               

        self.cfg.template_values['javaarch'] = javaarch                               
        self.cfg.generate_template_values()
```
Resulting in such an easyconfig file:
```
local_tarball_tmpl = '-'.join(['openjdk', '%%s(version)s_linux', '%s_bin.tar.gz'])
sources = [local_tarball_tmpl % '%(javaarch)s']
checksums = [
    {
        local_tarball_tmpl % 'x64':
            '91ac6fc353b6bf39d995572b700e37a20e119a87034eeb939a6f24356fbcd207',
        local_tarball_tmpl % 'aarch64':
            '3958f01858f9290c48c23e7804a0af3624e8eca6749b085c425df4c4f2f7dcbc',
    }
]

sanity_check_commands = ["java -version"]
```

Reply:
I don't know, what do you think?

Reply:
We actually handle the multiple architectures inside the wrapper package (see `Java-11.eb`). In principle, just having a single Java easyconfig for a particular version is also fine, but I think they have differed quite a lot in the past, e.g. Java 11.0.6 for PPC uses some type of fork it seems (different homepage and all), so perhaps it's best to just keep doing it like in Java 11?

We would need a Java-15.eb wrapper package regardless so things can avoid depending directly on 15.0.2.